### ISSUE TITLE
Relative paths are rewritten to absolute paths

### ISSUE THREAD

**Issue by wjhopper** [rstudio/rmarkdown#808]
I just discovered something unusual in how relative paths containing `../` are handled when rendering Rmarkdown documents. I also posted about this issue on [StackOverflow](http://stackoverflow.com/questions/39481092/how-to-stop-relative-paths-from-being-rewritten-to-absolute-paths-in-rmarkdown-d).

 Let's say I have a project with a structure as shown below:

```
./
├── files/
│   ├── some_image.jpg
│   └── test.Rmd
└── some_other_image.jpg
```

And I want to render the Rmarkdown document `test.Rmd` into HTML containing both jpegs `some_image.jpg` and `some_other_image.jpg`, like so:

```

---
title: "Path Test"
output:
  html_fragment:
    self_contained: false
    keep_md: true

---

Path without ../ is not altered
![Unaltered](some_image.jpg)

Path with ../ is converted to an absolute path!
![Altered](../some_other_image.jpg)
```

If I create the HTML document with `rmarkdown::render("files/test.Rmd", clean = FALSE)`, it produces the following HTML:

```
<p>Path without ../ is not altered <img src="some_image.jpg" alt="Unaltered" /></p>
<p>Path with ../ is converted to an absolute path! <img src="/home/will/test_project/some_other_image.jpg" alt="Altered" /></p>
```

If you look at the `src` attribute for each image, you can see that **the image path that was specified using `../` gets rewritten to use an absolute path!!** Moreover, this conversions _only_ happens when the specified file exists!

My questions: Why is this happening? This doesn't happen to me when manually using `pandoc` to convert this very same markdown to HTML? And if you inspect the intermediate markdown file produced, the path to `some_other_image.jpg` is still `../some_other_image.jpg`.

And what can I do to stop it? It works fine on my particular machine of course, but this is a big problem if the page is going to be deployed somewhere else.

I'm running Rmarkdown version 1.0 with R version 3.3.1 on debian testing, in case that's relevant for reproducing.

**Comment by wjhopper** [comment]
I dug into this  a bit more, and it appears the cause is the post-processing of the pandoc output. The path rewritting happens as a result of lines 155-167 in `html_document_base.R`

```r
    } else if (!self_contained) {
      # if we're not self-contained, find absolute references to the output
      # directory and replace them with relative ones
      image_relative <- function(img_src, src) {
        in_file <- utils::URLdecode(src)
        if (length(in_file) && file.exists(in_file)) {
          img_src <- sub(
            src, utils::URLencode(normalized_relative_to(output_dir, in_file)),
            img_src, fixed = TRUE)
        }
        img_src
      }
      output_str <- process_images(output_str, image_relative)
    }
```

The call to the function `normalized_relative_to` ends up rewriting the relative path to be an absolute one. I don't know what all the consequences of *not* calling this function here might be, but maybe one easy fix would to be allowing users to disable the post-processing of the pandoc output?

**Comment by Courvoisier13** [comment]
Hi, just posted a similar issue on Stackoverflow but with pdf files:
http://stackoverflow.com/questions/40470290/r-markdown-pdf-output-figure-with-out-width-90-generating-the-wrong-path-with

**Comment by yihui** [comment]
@wjhopper I propsed a fix at #1174. You may test it with `devtools::install_github('rstudio/rmarkdown#1174')`.

@Courvoisier13 Your issue is a different one. Usually I don't recommend using `output_dir` or `output_file`. They are known to be problematic, e.g. https://github.com/rstudio/rmarkdown/issues/587#issuecomment-168437646 #238 #348 #574 #1033 #861 etc. I believe the only reliable strategy is to render everything in the current directory, and move the output files later.
